nvflare/job_config/fed_job_config.py

```diff
@@ -396,23 +397,30 @@ def _get_args(self, component, custom_dir):
         args = {}
         if hasattr(component, "__dict__"):
             parameters = get_component_init_parameters(component)
-            attrs = component.__dict__
 
             for param in parameters:
-                attr_key = param if param in attrs.keys() else "_" + param
-
-                if attr_key in ["args", "kwargs"]:
+                if param in ["self", "args", "kwargs"]:
                     continue
 
-                if attr_key in attrs.keys() and parameters[param].default != attrs[attr_key]:
-                    if type(attrs[attr_key]).__name__ in dir(builtins):
-                        args[param] = attrs[attr_key]
-                    elif issubclass(attrs[attr_key].__class__, Enum):
-                        args[param] = attrs[attr_key].value
+                # Determine the correct attribute key by checking if the attribute actually exists
+                if hasattr(component, param):
```

Comment:
This set of code of determining attr_key is used in _add_referenced_components as well. Consider making it a reusable func.